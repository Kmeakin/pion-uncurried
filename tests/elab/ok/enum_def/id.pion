enum Id(A: Type) {
    MkId(_: A),
}

let mkid = MkId;

let get
= fn(B: Type, it: Id(B)) => match it {
    MkId(z) => z,
};

let get_ann:
fn (C: Type, that: Id(C)) -> C
= fn(B: ?_, it: ?_) => match it {
    MkId(z) => z,
};

enum Id2(A: Type) {
    MkId2(_: A, _: A),
}

let mkid2 = MkId2;

let get2
= fn(B: Type, it: Id2(B)) => match it {
    MkId2(z,foo) => z,
};

let get2_ann:
fn (C: Type, that: Id2(C)) -> C
= fn(B: ?_, it: ?_) => match it {
    MkId2(z,foo) => z,
};