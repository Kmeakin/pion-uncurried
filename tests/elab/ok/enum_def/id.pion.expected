enum Id(A: Type): Type {
    MkId(_: A): Id(A),
}

let mkid: fn(A: Type, _: A) -> Id(A) = MkId;

let get: fn(B: Type, it: Id(B)) -> B = fn(B: Type, it: Id(B)) => match it {
    MkId(z) => z,
};

let get_ann: fn(C: Type, that: Id(C)) -> C = fn(B: Type, it: Id(B)) => match it {
    MkId(z) => z,
};

enum Id2(A: Type): Type {
    MkId2(_: A, _: A): Id2(A),
}

let mkid2: fn(A: Type, _: A, _: A) -> Id2(A) = MkId2;

let get2: fn(B: Type, it: Id2(B)) -> B = fn(B: Type, it: Id2(B)) => match it {
    MkId2(z, foo) => z,
};

let get2_ann: fn(C: Type, that: Id2(C)) -> C = fn(B: Type, it: Id2(B)) => match it {
    MkId2(z, foo) => z,
};
